benchexec/tools/cbmc.py

```diff
@@ -102,12 +108,15 @@ def isErrorMessage(msg):
 
         return status
 
-    def determine_result(self, returncode, returnsignal, output, isTimeout):
+    def determine_result(self, run):
+        returnsignal = run.exit_code.signal or 0
```

Comment:
The fact that previously `returnsignal` and `returncode` were 0 by default is actually quite bad API and makes the code complex to understand (because for example `returncode==0` means that the tool exited successfully, but only if `returnsignal` is not `None` at the same time). In general, handling the new values (with default `None`) is easier and less errorprone. Thus I would suggest to refactor the code to directly use the fields of `run.exit_code` as they are, preferably even by passing the `ProcessExitCode` instance around instead of individual values.

Reply:
fixed for value in 5823bfb9ed00b07a19413ae454100020ee3a5bfa